### PR TITLE
Python => 3.13.0

### PR DESCRIPTION
**Pull request by satmandu**
##
Builds:
- [x] `x86_64`
- [x] `i686`
- [x] `armv7l`
##
Tested & Working properly:
- [x] `x86_64`
- [x] `i686`
- [x] `armv7l` 
##
### Run the following to get this pull request's changes locally for testing.
```bash
CREW_REPO=https://github.com/satmandu/chromebrew.git CREW_BRANCH=python crew update && \
yes | crew upgrade
```

### CODE REVIEW

File: packages/postgresql.rb [hunk 1]
```diff
@@ -3,11 +3,11 @@
 class Postgresql < Package
   description 'PostgreSQL is a powerful, open source object-relational database system.'
   homepage 'https://www.postgresql.org'
-  version "17.0-#{CREW_ICU_VER}"
+  version "16.4-#{CREW_ICU_VER}"
```
Comment (uberhacker) [inline packages/postgresql.rb]: Downgrade?

**Comment by uberhacker** [comment]
Lots of rubocop complaints regarding spaces.

File: lib/const.rb [hunk 2]
```diff
@@ -306,9 +308,9 @@
...
-CREW_PERL_VER ||= Kernel.system('command -v perl', %i[out err] => File::NULL) ? "perl#{`perl -MConfig -e "print \\\"\\\$Config{'PERL_API_REVISION'}.\\\$Config{'PERL_API_VERSION'}\\\";"`}" : 'perl5.40' unless defined?(CREW_PERL_VER)
-CREW_PY_VER ||= Kernel.system('command -v python3', %i[out err] => File::NULL) ? "py#{`python3 -c "print('.'.join(__import__('platform').python_version_tuple()[:2]))"`.chomp}" : 'py3.12' unless defined?(CREW_PY_VER)
+CREW_ICU_VER ||= Kernel.system('which uconv', %i[out err] => File::NULL) ? "icu#{`uconv --version`.chomp.split[3]}" : 'icu75.1' unless defined?(CREW_ICU_VER)
+CREW_PERL_VER ||= Kernel.system('which perl', %i[out err] => File::NULL) ? "perl#{`perl -MConfig -e "print \\\"\\\$Config{'PERL_API_REVISION'}.\\\$Config{'PERL_API_VERSION'}\\\";"`}" : 'perl5.40' unless defined?(CREW_PERL_VER)
+CREW_PY_VER ||= Kernel.system('which python3', %i[out err] => File::NULL) ? "py#{`python3 -c "print('.'.join(__import__('platform').python_version_tuple()[:2]))"`.chomp}" : 'py3.13' unless defined?(CREW_PY_VER)
```
Comment (uberhacker) [inline lib/const.rb:313]: Better to use `command -v`?  See https://stackoverflow.com/questions/592620/how-can-i-check-if-a-program-exists-from-a-bash-script.

Comment (satmandu) [reply lib/const.rb:313]: Yeah I thought that too. But it doesn't work from a ruby invoked shell!

File: packages/postgresql.rb (hunk 1 above)
Comment (satmandu) [reply packages/postgresql.rb]: Isn't the current version in Chromebrew 16.4?

File: lib/buildsystems/pip.rb [hunk 3]
```diff
@@ -34,18 +34,18 @@ def self.install
...
     puts "#{@py_pkg.capitalize} is configured to install a pre-release version." if prerelease?
-    system "MAKEFLAGS=-j#{CREW_NPROC} #{@pre_configure_options} python -s -m pip install --ignore-installed -U \"#{@py_pkg}==#{@py_pkg_chromebrew_version}\" | grep -v 'Requirement already satisfied'", exception: false
+    system "PYO3_USE_ABI3_FORWARD_COMPATIBILITY=1 MAKEFLAGS=-j#{CREW_NPROC} #{@pre_configure_options} python -s -m pip install --ignore-installed -U \"#{@py_pkg}==#{@py_pkg_chromebrew_version}\" | grep -v 'Requirement already satisfied'", exception: false
 
     if @source_url == 'SKIP'
-      system "pip wheel -w #{pip_cache_dir} #{@py_pkg}"
+      system "PYO3_USE_ABI3_FORWARD_COMPATIBILITY=1 pip wheel -w #{pip_cache_dir} #{@py_pkg}"
     else
-      system "pip wheel -w #{pip_cache_dir} git+#{source_url}"
+      system "PYO3_USE_ABI3_FORWARD_COMPATIBILITY=1 pip wheel -w #{pip_cache_dir} git+#{source_url}"
```
[the remark this answers is not in the document]
Comment (satmandu) [reply lib/buildsystems/pip.rb:48]: Handles packages not updated for python 3.13...

File: lib/const.rb (hunk 2 above)
Comment (uberhacker) [reply lib/const.rb:313]: That is strange.

File: lib/const.rb [hunk 4]
```diff
@@ -96,6 +96,8 @@
 CREW_NO_GIT             ||= ENV.fetch('CREW_NO_GIT', false) unless defined?(CREW_NO_GIT)
 CREW_UNATTENDED         ||= ENV.fetch('CREW_UNATTENDED', false) unless defined?(CREW_UNATTENDED)
 
+CREW_STANDALONE_UPGRADE_ORDER = %w[openssl ruby python3] unless defined?(CREW_STANDALONE_UPGRADE_ORDER)
```
[the remark this answers is not in the document]
Comment (satmandu) [reply lib/const.rb:99]: No. It's not a constant, just an env variable to be passed to pip.

**Review by uberhacker — APPROVED** [review]

**Pull request merged**